practice-one/src/components/Form/index.tsx

```diff
@@ -0,0 +1,184 @@
+
+
+interface StudentFormProps {
+    closeForm: () => void;
+    onStudentAdd: () => void;
+}
+
+interface IFormInput {
+    name: string;
+    email: string;
+    phone: string;
+    gender: string;
+    password: string;
+    classes: string;
+}
+
+const StudentForm: React.FC<StudentFormProps> = ({ closeForm, onStudentAdd }) => {
+    const {
+        register,
+        handleSubmit,
+        formState: { errors }
+    } = useForm<IFormInput>();
+    const fileInputRef = useRef<HTMLInputElement>(null);
+    const [imageFile, setImageFile] = useState<File | null>(null);
+    const [imagePreviewUrl, setImagePreviewUrl] = useState<string | null>(null);
+
+    const onSubmit = async (data: IFormInput) => {
+        console.log("Form Submitted", data);
+        let uploadedImageUrl = null;
+        if (imageFile) {
+            const uploadResult = await uploadImage(imageFile);
+    
+            if (uploadResult.error) {
+                console.error("Error uploading image:", uploadResult.error);
+            } else {
+                uploadedImageUrl = uploadResult.data;
+            }
+        }
+    
+        const newStudentData = {
+            ...data,
+            studentID: Math.floor(10000 + Math.random() * 90000),
+            studentAge: "17",
+            imageUrl: uploadedImageUrl
+        };
+
+        const students = JSON.parse(localStorage.getItem('students') || '[]');
+        students.push(newStudentData);
+        localStorage.setItem('students', JSON.stringify(students));
+        onStudentAdd();
+        closeForm();
+    };
+
+    const handleImageChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        const file = event.target.files && event.target.files[0];
+        if (file) {
+            setImageFile(file);
+            setImagePreviewUrl(URL.createObjectURL(file));
+        }
+    };
+
+    const genderOptions = [
+        { value: 'Male', label: 'Male' },
+        { value: 'Female', label: 'Female' },
+    ];
+
+    const classOptions = [
+        { value: 'SS1', label: 'SS1' },
+        { value: 'SS2', label: 'SS2' },
+        { value: 'SS3', label: 'SS3' },
+        { value: 'SS4', label: 'SS4' },
+        { value: 'SS5', label: 'SS5' },
+    ];
+
+    return (
+        <form onSubmit={handleSubmit(onSubmit)} className="add-student-form">
+            <div className="form-header">
+                <h2 className="heading-form">Add Student</h2>
+                <div className="link-group">
+                    <a href="#" className="link-group-item">Manually</a>
+                    <a href="#" className="link-group-item">Import CSV</a>
+                </div>
+            </div>
+            <div className="form-body">
+                <div className="form-body-item">
+
+                    <div className="student-name item">
+                        <p className="item-title">Name</p>
+                        <Input
+                            {...register("name", { required: "Name is required" })}
```

Comment:
Can you try to use <Controller /> of react-hook-form it help you on this case no need to use register